docs/MEMORY-SETTINGS.md

```diff
@@ -0,0 +1,495 @@
+
+Increase `ctrd_mem` if you observe that the containerd service is reaching its
+memory limits. This can be identified by instances where the Out-Of-Memory (OOM)
+killer is fired with constraint `CONSTRAINT_MEMCG` and `oom_memcg` values set to
+the `eve/containerd` cgroup:
+
+```text
+oom-kill:constraint=CONSTRAINT_MEMCG,...,oom_memcg=/eve/containerd,...
+```
+
+Don't forget to increase `dom0_mem` value as well, as the `eve/containerd` cgroup
+is a child of the `eve` cgroup.
+
+#### When to decrease `ctrd_mem`?
+
+Decrease `ctrd_mem` when you need to decrease the value of `dom0_mem`. See the
+section "When to decrease `dom0_mem`?" for more details.
+
+#### Impact of setting `ctrd_mem` too high
+
+In the simplest case setting `ctrd_mem` too high leads to too high value of
+`dom0_mem` (the `eve` cgroup), which can lead to inefficient use of system
+resources, where memory is reserved and not available for user applications.
+
+See the section "Impact of setting `dom0_mem` too high" for more details.
+
+#### Impact of setting `ctrd_mem` too low
+
+Setting `ctrd_mem` too low could cause the containerd service to frequently hit
+memory limits, triggering the OOM killer within the `eve/containerd` cgroup.
+
+See the section "When to increase `ctrd_mem`?" for more details.
+
+### VMM Overhead
+
+The VMM overhead is the extra memory that is used by the system to run the app
+instance. It is calculated automatically in runtime, but there is a possibility
+to set it manually. It can be useful when the device has little memory and
+the automatic calculation can be too high.
+
+#### When to increase VMM Overhead?
+
+You definitely should increase the VMM overhead if you observe the OOM killer
+firing with constraint `CONSTRAINT_MEMCG` and the `oom_memcg` value set to one
+of the cgroups of the user applications:
+
+```text
+oom-kill:constraint=CONSTRAINT_MEMCG,...,oom_memcg=/eve-user-apps/<uuid>,...
+```
+
+It means that the user application (the QEMU process) is reaching its memory
+limits on the host.
+
+#### When to decrease VMM Overhead?
+
+Decrease the VMM overhead if you need to free up memory for the EVE services,
+for example, if you need to give more memory to the EVE services. See the
+sections "When to increase `dom0_mem`?", "When to increase `eve_mem`?", and
+"When to increase `ctrd_mem`?" for more details.
+
+Also, decreasing the VMM overhead can be useful if you want to try to run more
+user applications on the device. The estimated VMM overhead can be too high in
+some cases, and if you run several user applications on the device, it can lead
+to a lot of memory being "wasted" and not available for the EVE services or
+other user applications.
+
+An indication that the EVE services require more memory can be the situation
+when the EVE services are frequently hitting memory limits and the OOM killer is
+fired with constraint `CONSTRAINT_NONE` and the `oom_memcg` value set to one of
+the cgroups of the EVE services:
```

Comment:
You probably forgot to list cgroups after the colon or the sentence should end with a fullstop.

Reply:
ooops, yep... Will create a new PR soon